### ISSUE TITLE
Could auth response differentiate invalid credentials?

### ISSUE THREAD

**Issue by wenzowski** [KnowledgeGarden/BacksideServletKS#6]
Noticed that if we attempt to submit an authentication request with invalid credentials we don't receive an `rToken`, which makes sense. However, since `rMsg` is `''` instead of `'ok'` it can be difficult to differentiate between an invalid request and a failed authentication.

Perhaps a request with invalid credentials could result in the following response instead?

``` json
{
  "rMsg": "ok",
  "rToken": ""
}
```

**Comment by KnowledgeGarden** [comment]
That may be a bug. There is a design decision that only really foul (whatever that means) errors at the server would return something other than 'ok'.  As I recall, an early reason for that was in my Meteor frontside code where anything other than 'ok' threw Meteor into what I call "Oh Shit!" mode.  It seems worth doing a recursive grep through the code for rMessage = "" (where rMessage may be the String variable which becomes rMsg in JSON.

**Comment by wenzowski** [comment]
Well...[I can see it just needs](https://github.com/KnowledgeGarden/BacksideServletKS/blob/master/src/main/java/org/topicquests/backside/servlet/apps/usr/persist/H2UserDatabase.java#L92) a `result.addErrorString("Invalid Credentials");` somewhere in there, but I'm getting lost in the single letter variables. What is `rs.next()`, and how would we know if h2 didn't find a match?

**Comment by OpenSherlock** [comment]
Yes indeed. the platform just returns null if it doesn't authenticate. Lots of room for evolution.

**Comment by wenzowski** [comment]
Would changing this return value negatively affect TQPortalKS? Unless a failed auth is the only case where it could ever return null I'd prefer to be as explicit as possible if we can.

**Comment by KnowledgeGarden** [comment]
It's not hard to tweak TQPortal to deal with a different return value. As I
recall, it also looks for error messages; the IReturn concept  grew out of
transliterating PHP to Java and the need to return lots of stuff in one
object.

**Comment by wenzowski** [comment]
Great. Looking a little closer at my moment of confusion above, a `java.sql.ResultSet.next()` ought to throw a `java.sql.SQLException`, which you're `catch()`ing.

**Comment by OpenSherlock** [comment]
Yes. Where a class method would typically be expected to throw an exception, in most, but not all such cases, I wrap a try-catch when I am returning an IResult which is implemented as ResultPojo. That same object carries two resultObjects; most methods just use one, but some might use both. The canonical use is result.getReturnObject();
